app/components/transUnit/TransUnitService.js

```diff
@@ -190,7 +237,9 @@
     }
 
     function setSelected(transUnitCtrl, isSelected) {
-      transUnitCtrl.selected = isSelected || false;
+      if(transUnitCtrl.selected !== isSelected) {
+        transUnitCtrl.selected = isSelected || false;
+      }
```

Comment:
Is there a problem with just setting the value to the same value? The extra check makes the code more complicated so I would like to know the reason for it, and we had better add a comment indicating what the reason is or someone might try to simplify it back to the previous structure later.